drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -709,6 +712,270 @@ static int siwx91x_stats(const struct device *dev, struct net_stats_wifi *stats)
 }
 #endif
 
+struct siwx91x_ps_profile {
+	/* HIGH_PERFORMANCE by default */
+	sl_si91x_performance_profile_t req_ps_mode;
+	uint16_t listen_interval;
+} siwx91x_ps_cfg;
+
+static int siwx91x_get_beacon_interval(const struct device *dev)
+{
+	sl_wifi_operational_statistics_t sl_stat = { };
+	struct siwx91x_dev *sidev = dev->data;
+	sl_status_t status;
+
+	if (sidev->state != WIFI_STATE_COMPLETED) {
+		/* fail reason */
+		LOG_ERR("Device not connected");
+		return -EIO;
+	}
+
+	status = sl_wifi_get_operational_statistics(SL_WIFI_CLIENT_INTERFACE, &sl_stat);
+	if (status != SL_STATUS_OK) {
+		LOG_ERR("Failed to get beacon interval: 0x%x", status);
+		return -EINVAL;
+	}
+
+	return sys_get_le16(sl_stat.beacon_interval);
```

Comment:
Until now, we didn't considered the host CPU could be big endian. I believe it does not make sense to do it for `sl_stat.beacon_interval`.